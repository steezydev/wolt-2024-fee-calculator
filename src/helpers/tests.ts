const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
