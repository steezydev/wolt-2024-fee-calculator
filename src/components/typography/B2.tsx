import { classNames } from '@/helpers/classNames';
import { TypographyProps } from '@/types/props/TypographyProps';

const B2 = ({ children, className }: TypographyProps) => {
  return (
    <p className={classNames('text-lg text-black dark:text-white', className)}>
      {children}
    </p>
  );
};

export default B2;
