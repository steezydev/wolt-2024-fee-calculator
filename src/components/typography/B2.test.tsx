import B2 from './B2';
import { typographyRenderTests } from './Typography.tests';

typographyRenderTests('B2', B2);
