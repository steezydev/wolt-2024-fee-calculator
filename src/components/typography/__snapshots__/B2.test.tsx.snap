// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`B2 when asked to render  renders 1`] = `
<p
  className="text-lg text-black dark:text-white"
>
  Hello world!
</p>
`;

exports[`B2 when asked to render with additional class  renders with additional class 1`] = `
<p
  className="text-lg text-black dark:text-white some-class"
>
  Hello world!
</p>
`;
